packages/airnode-adapter/src/response-processing/encoding.ts

```diff
@@ -21,12 +21,26 @@ export function getSolidityType(type: ResponseType) {
 
 export function encodeValue(value: ValueType, type: ResponseType): string {
   const solidityType = getSolidityType(type);
-
-  return ethers.utils.defaultAbiCoder.encode([solidityType], [value]);
+  try {
+    return ethers.utils.defaultAbiCoder.encode([solidityType], [value]);
+  } catch (e) {
+    if (e instanceof Error) {
+      throw new EncodingError((e as Error).message);
+    } else {
```

Comment:
Nit: there's no need for the else block here. It could be written as

```ts
if (e instanceof Error) {
  throw new EncodingError(...);
}
throw new EncodingError(...);
```

Reply:
In general, the more right indents, the harder the code becomes to read (at least in my opinion 😁 ). If you return (or throw) from inside the `if` then you don't need the `else`. A side effect of this is that you're often forced to write smaller functions. This is why you'll find very few `else` blocks in the Airnode repo.

Reply:
I will remove the `else` blocks :+1: